### ISSUE TITLE
RC4 enabled on SSL mosquitto

### ISSUE THREAD

**Issue by nikhilve99** [eclipse/mosquitto#541]
Hi team,

I have just configured ssl on port 443 on my mqtt server, after penetration on my server i am getting RC4 enabled on my server. 
how i can disable RC4 on mqtt server.
kindly help

**Comment by toast-uz** [comment]
Use ciphers directive in mosquitto.conf as follows:
ciphers ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384

Although it is a little tough, you must list up all of ciphers you want to enable.

**Comment by nikhilve99** [comment]
Plz direct me. I want to disable RC4 cipher through mosquitto.conf.

Regards
Nikhil

**Comment by toast-uz** [comment]
Seems no way to disable specific ciphers. You must list up all of ciphers you want to keep. You can know all of ciphers list by "openssl ciphers" and remove something which include RC4 from the list. Attention too long (> 1024 bytes) line of mosquitto.conf may not be read properly.

**Comment by nikhilve99** [comment]
Thanks for your reply.
Actually what issue I am facing , one of our customer did penetration on
our Mqtt server and found RC4 enabled. That's why any how I need disable
RC4 from ssl .

Plz suggest what I can do in this situation.

Regards
Nikhil

**Comment by toast-uz** [comment]
Anyway, it's not an issue of mosquitto.
Close this issue and ask on mosquitto-dev ML.
If you can show your mosquitto.conf (mask your global IP and any specifying you), you can get information more practically.

**Comment by toast-uz** [comment]
Additional one comment I can post now.
I guess disabling RC4 is not always required. Ordinary, server prioritizes low for RC4. So the right to choice RC4 and risk of using RC4 belong to client. 
For web servers, disabling RC4 is right because of every browser gave up to use RC4. But for mqtt servers, disabling RC4 brings the trouble for old clients keep using RC4. You might cut out your old customers by disabling RC4.
In conclusion, you have choice to declare that keeping RC4 enabled is proper for mqtt servers.

**Comment by toast-uz** [comment]
Of course, you cannot declare if this is a dedicated server for the new customer. :)

**Comment by toast-uz** [comment]
@nikhilve99 do you have any more information?

**Comment by toast-uz** [comment]
See also #652 .

**Comment by toast-uz** [comment]
Closing as no information for a long time.
